build-tools/packages/build-cli/src/commands/release/prepare.ts

```diff
@@ -0,0 +1,266 @@
+	// the client release group, we can't easily scope it to just the client. Thus, we always run it at the root just
+	// like we do in CI.
+	await execa.command(`npm run policy-check:asserts`, {
+		cwd: context.gitRepo.resolvedRoot,
+	});
+
+	// check for policy check violation
+	const afterPolicyCheckStatus = await context.gitRepo.getStatus();
+	if (afterPolicyCheckStatus !== "") {
+		const repo = new Repository({ baseDir: context.repo.resolvedRoot });
+		await repo.gitClient.reset(ResetMode.HARD);
+		return {
+			result: false,
+			message: "Found some untagged asserts. These should be tagged before release.",
+			fixCommand: "pnpm run policy-check:asserts",
+		};
+	}
+
+	return { result: true };
+}
+
+/**
+ * All the checks that should be executed. The checks are executed serially in this order.
+ */
+const allChecks: ReadonlyMap<string, CheckFunction> = new Map([
+	["Branch has no local changes", checkNoLocalChanges],
+	["Has a microsoft/FluidFramework remote", checkHasRemote],
+	["The local branch is up to date with the remote", checkBranchUpToDate],
+	["Dependencies are installed locally", checkDepsInstalled],
+	["Has no pre-release Fluid dependencies", checkHasNoPrereleaseDependencies],
+	["No repo policy violations", checkPolicy],
+	["No untagged asserts", checkAsserts],
+]);
+
+export class ReleasePrepareCommand extends BaseCommand<typeof ReleasePrepareCommand> {
+	static readonly summary =
+		`Runs checks on a local branch to verify it is ready to serve as the base for a release branch.`;
+
+	static readonly description = `Runs the following checks:\n\n- ${[...allChecks.keys()].join(
+		"\n- ",
+	)}`;
+
+	static readonly args = {
+		package_or_release_group: packageOrReleaseGroupArg({
+			description:
+				"The name of a package or a release group. Defaults to the client release group if not specified.",
+			default: "client",
+		}),
+	};
+
+	async run(): Promise<void> {
+		const context = await this.getContext();
+
+		const rgArg = this.args.package_or_release_group;
+		const pkgOrReleaseGroup = findPackageOrReleaseGroup(rgArg, context);
+		if (pkgOrReleaseGroup === undefined) {
+			this.error(`Can't find package or release group "${rgArg}"`, { exit: 1 });
+		}
+		this.verbose(`Release group or package found: ${pkgOrReleaseGroup.name}`);
+
+		this.logHr();
+		for (const [name, check] of allChecks) {
+			// eslint-disable-next-line no-await-in-loop
+			const checkResult = await check(context, pkgOrReleaseGroup, this.flags.verbose);
+			const { result, message, fixCommand, fatal } = checkResult;
+			this.log(
+				`${result ? chalk.bgGreen.black(" ✔︎ ") : chalk.bgRed.white(" ✖︎ ")} ${
+					message === undefined ? name : chalk.red(message)
+				}`,
+			);
+			if (fixCommand !== undefined) {
+				this.logIndent(
+					`${chalk.yellow(`Possible fix command:`)} ${chalk.yellow.bold(fixCommand)}`,
+					6,
+				);
+			}
+			if (fatal === true) {
+				this.error("Can't do other checks until the failures are resolved.", { exit: 5 });
+			}
+		}
```

Comment:
I noticed this entire file has no tests. While a lot of it is hard to test, I think a lot of the logic in this function could be tested if you really want to, this section of logic could be factored into its own function (passed in allChecks, and pkgOrReleaseGroup). That function could then be tested with a couple combinations of checks (ex: one that just errors, ones that pass etc), allowing you to check that the correct log output is generated for fatal and non fatal errors for example, which might not be the easiest to cover with manual testing when working on this code in the future.